src/message_interface/message_handler.rs

```diff
@@ -330,6 +330,34 @@ impl WalletMessageHandler {
                 })
                 .await
             }
+            #[cfg(feature = "participation")]
+            Message::GetParticipationEvent(event_id) => {
+                convert_async_panics(|| async {
+                    let event = self
+                        .account_manager
+                        .get_participation_event(event_id)
+                        .await?
+                        .map(|(e, _)| e);
+                    Ok(Response::ParticipationEvent(event))
+                })
+                .await
+            }
+            #[cfg(feature = "participation")]
+            Message::GetParticipationEventStatus(event_id) => {
+                convert_async_panics(|| async {
+                    let event_status = self.account_manager.get_participation_event_status(event_id).await?;
```

Comment:
I know I previously approved this but I also think this should return Option and not an error if the record is not found

Reply:
But this is the event status which gets requested from the node

Reply:
Not sure what you mean. What happens if you pass a bullshit event ID?

Reply:
Then it should return an error, that the event wasn't found and not None

Reply:
Why does `get_participation_event` return an option and not `get_participation_event_status`? They are doing pretty similar stuff ?

Reply:
get_participation_event works with local data from the db, get_participation_event_status returns what the node returns for this event

Reply:
Ok I see. I now remember asking in the past why we don't return Option for the routes, this still kinda puzzles me a bit. But fine 👍🏻